Add option to disable perl

Introduces the --disable-perl option and skips installation of some Perl
dependencies if the option is set.

crbug.com/877675.


diff --git a/buildtools/wafsamba/wscript b/buildtools/wafsamba/wscript
index 3b36b57..0f2c7a6 100644
--- a/buildtools/wafsamba/wscript
+++ b/buildtools/wafsamba/wscript
@@ -196,6 +196,10 @@ def set_options(opt):
                    help='tag release in git at the same time',
                    type='string', action='store', dest='TAG_RELEASE')
 
+    opt.add_option('--disable-perl',
+                    help='do not generate perl modules',
+                    action='store_true', dest='disable_perl', default=False)
+
     opt.add_option('--disable-python',
                     help='do not generate python modules',
                     action='store_true', dest='disable_python', default=False)
@@ -285,6 +289,7 @@ def configure(conf):
     conf.env.AUTOCONF_HOST  = Options.options.AUTOCONF_HOST
     conf.env.AUTOCONF_PROGRAM_PREFIX = Options.options.AUTOCONF_PROGRAM_PREFIX
 
+    conf.env.disable_perl = Options.options.disable_perl
     conf.env.disable_python = Options.options.disable_python
 
     conf.env.EXTRA_PYTHON = Options.options.EXTRA_PYTHON
diff --git a/pidl/lib/wscript_build b/pidl/lib/wscript_build
index 54b3170..328ba21 100644
--- a/pidl/lib/wscript_build
+++ b/pidl/lib/wscript_build
@@ -1,37 +1,38 @@
 #!/usr/bin/env python
 
-# install the pidl modules
-bld.INSTALL_FILES(bld.env.PERL_LIB_INSTALL_DIR,
-                  '''
-                  Parse/Pidl.pm
-                  Parse/Pidl/Samba4.pm
-                  Parse/Pidl/CUtil.pm
-                  Parse/Pidl/Expr.pm
-                  Parse/Pidl/Wireshark/Conformance.pm
-                  Parse/Pidl/Wireshark/NDR.pm
-                  Parse/Pidl/ODL.pm
-                  Parse/Pidl/Dump.pm
-                  Parse/Pidl/Util.pm
-                  Parse/Pidl/Samba4/Header.pm
-                  Parse/Pidl/Samba4/COM/Header.pm
-                  Parse/Pidl/Samba4/COM/Proxy.pm
-                  Parse/Pidl/Samba4/COM/Stub.pm
-                  Parse/Pidl/Samba4/TDR.pm
-                  Parse/Pidl/Samba4/NDR/Server.pm
-                  Parse/Pidl/Samba4/NDR/Client.pm
-                  Parse/Pidl/Samba4/NDR/Parser.pm
-                  Parse/Pidl/Samba4/Python.pm
-                  Parse/Pidl/Samba4/Template.pm
-                  Parse/Pidl/IDL.pm
-                  Parse/Pidl/Typelist.pm
-                  Parse/Pidl/Samba3/ClientNDR.pm
-                  Parse/Pidl/Samba3/ServerNDR.pm
-                  Parse/Pidl/Compat.pm
-                  Parse/Pidl/NDR.pm
-                  ''',
-                  flat=False)
+if not bld.env.disable_perl:
+	# install the pidl modules
+	bld.INSTALL_FILES(bld.env.PERL_LIB_INSTALL_DIR,
+			'''
+			Parse/Pidl.pm
+			Parse/Pidl/Samba4.pm
+			Parse/Pidl/CUtil.pm
+			Parse/Pidl/Expr.pm
+			Parse/Pidl/Wireshark/Conformance.pm
+			Parse/Pidl/Wireshark/NDR.pm
+			Parse/Pidl/ODL.pm
+			Parse/Pidl/Dump.pm
+			Parse/Pidl/Util.pm
+			Parse/Pidl/Samba4/Header.pm
+			Parse/Pidl/Samba4/COM/Header.pm
+			Parse/Pidl/Samba4/COM/Proxy.pm
+			Parse/Pidl/Samba4/COM/Stub.pm
+			Parse/Pidl/Samba4/TDR.pm
+			Parse/Pidl/Samba4/NDR/Server.pm
+			Parse/Pidl/Samba4/NDR/Client.pm
+			Parse/Pidl/Samba4/NDR/Parser.pm
+			Parse/Pidl/Samba4/Python.pm
+			Parse/Pidl/Samba4/Template.pm
+			Parse/Pidl/IDL.pm
+			Parse/Pidl/Typelist.pm
+			Parse/Pidl/Samba3/ClientNDR.pm
+			Parse/Pidl/Samba3/ServerNDR.pm
+			Parse/Pidl/Compat.pm
+			Parse/Pidl/NDR.pm
+			''',
+			flat=False)
 
-if not bld.CONFIG_SET('USING_SYSTEM_PARSE_YAPP_DRIVER'):
-    bld.INSTALL_FILES(bld.env.PERL_LIB_INSTALL_DIR,
-                      'Parse/Yapp/Driver.pm',
-                      flat=False)
+	if not bld.CONFIG_SET('USING_SYSTEM_PARSE_YAPP_DRIVER'):
+		bld.INSTALL_FILES(bld.env.PERL_LIB_INSTALL_DIR,
+				'Parse/Yapp/Driver.pm',
+				flat=False)
